sections/advanced/email-testing.md

```diff
@@ -0,0 +1,213 @@
+
+Some ways to address unique users is by utilizing [gmail tricks](https://www.idownloadblog.com/2018/12/19/gmail-email-address-tricks/) or [AWS Simple Email Service](https://docs.aws.amazon.com/ses/latest/DeveloperGuide/send-email-simulator.html). If these work for your needs - you may not have to check actual email content - then you are on the right path with stateless tests. 
+However, these approaches can be problematic; for example non-existing emails can prompt bouncing emails to your cloud service and that can be a headache. If you want to avoid such issues and check real email content in automation, email services provide value. 
+
+Email services can also provide cost savings in text execution time by receiving emails faster, tests running quicker in the pipeline with less CI resources being consumed and less time waiting for tests to finish. If you are running 1000 pipelines a year, and save 3-4 seconds per pipeline execution, the email service can be already paying for its annual subscription just by providing extra speed.
+
+
+### Achieving Stateless tests with unique emails
+
+If in every test execution a new, unique user was used and the emails to this unique user could be verified in isolation, it would be possible to achieve a stateless test. The only side effect would be to the email service inbox, but if the test only checked emails by reference and cleaned up after itself, the email service mailbox would not be impacted.
+
+This is easy to achieve with Mailosaur, here are two approaches to this ; [Mailosaur's Node package](https://www.npmjs.com/package/mailosaur) or our own util with `faker.js`.
+
+```javascript
+// at cypress/plugins/mailosaur-tasks.js
+
+// generates a random email address
+// sample output:   ojh788.<serverId>@mailosaur.io
+const createEmail = () => mailosaurClient
+  .servers
+  .generateEmailAddress(envVars.MAILOSAUR_SERVERID);
+);
+
+// our custom function at a helper file or commands file. The only difference is the defined prefixed name.
+// sample output:  fakerJsName.<serverId>@mailosaur.io
+const createMailosaurEmail = randomName => 
+  `${randomName}.${Cypress.env('MAILOSAUR_SERVERID')}@mailosaur.io`;
+```
+
+<br/><br/>
+
+## (2) Explanation - what to test in an email and how
+
+First, let's elaborate on the setup we need.
+
+### Test Setup and Hybrid Approach: [Mailosaur Rest API](https://docs.mailosaur.com/reference) with `cy.request()` and [Mailosaur's Node package](https://www.npmjs.com/package/mailosaur) with `cy.task()`
+
+Mailosaur provides an [npm package](https://www.npmjs.com/package/mailosaur) and effectively all the Node code samples in the [api documentation](https://docs.mailosaur.com/reference) can be converted to `cy.task()`. Another approach is to implement Mailosaur's Rest API 'from scratch with `cy.request()`.
```

Comment:
I’d maybe add a screenshot or a simple explanation of how Mailosaur works (it creates a server for you, which you can access via API or UI. You can create multiple servers, search for messages, attachments, content of the email. If using API, the content of the message (links, attachments, to, cc, bcc) are parsed into json.

Reply:
Oh sorry, I just realized there is an example on how to test email content at the bottom 🤦